applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -162,6 +162,10 @@ public function discussion($CategoryUrlCode = '') {
             $this->title(t('New Discussion'));
         }
 
+        if (isset($this->Draft) && $this->Draft->Type) {
```

Comment:
Actually, what about modifying the touchValue line under that to: `touchValue('Type', $this->Data, val('Type', $this->Draft, 'Discussion'));`?